luxos.py

```diff
@@ -512,6 +529,14 @@ def execute_command(host: str, port: int, timeout_sec: int, cmd: str,
             for thread in threads:
```

Comment:
Why do we need to wait for all the threads to finish to continue with the rest? Imagine we have a batch of 100 miners, and 2 of them are somehow blocking the process, this will wait until those two finish while we can process another 98 as we have 98 free slots.

Much better approach will be to use a FIFO queue and limit it to max_threads.

Reply:
As reviewed with Guz, we will come back to this helper file later, so approving as delay addition is ok.